ANDAUTH-82 Add Landscape mode for SyncSettingsActivity

somenath1435 opened the pull request:
Before:

![Screenshot_20200313-043843 1](https://user-images.githubusercontent.com/43093724/76575420-10c73300-64e5-11ea-9151-da5f53c95800.png)

After:

![Screenshot_20200313-042833 1](https://user-images.githubusercontent.com/43093724/76575438-1f154f00-64e5-11ea-9c8f-40d2c8305c22.png)

somenath1435 committed:
ANDAUTH-82 Add Landscape mode for SyncSettingsActivity

InsanusMokrassar reviewed (changes requested):
I think, that we can place the whole account button on the right side, replace save button by a float button (info you can read [here](https://material.io/components/buttons-floating-action-button/)). So, the order will be next:

about version | account username |
-------------------| ------------- |
sync type        | account server address |

------------------------------------------------------

list of accounts
save fab button

somenath1435 committed:
added FAB and made few changes

somenath1435 commented:
Modified: 
![Screenshot_20200314-035438 1](https://user-images.githubusercontent.com/43093724/76664102-34ec4800-65a9-11ea-8dc0-cd0e04c28e50.png)

somenath1435 commented:
@InsanusMokrassar suggestions please for this PR

InsanusMokrassar commented on app/src/main/java/org/xwiki/android/sync/activities/SyncSettingsActivity.kt:
```diff
@@ -185,7 +187,13 @@ class SyncSettingsActivity : AppCompatActivity(), GroupsListChangeListener {
-        layoutManager = LinearLayoutManager(this)
+        if(getScreenOrientation(this)=="SCREEN_ORIENTATION_LANDSCAPE" ||
+            getScreenOrientation(this)=="SCREEN_ORIENTATION_REVERSE_LANDSCAPE"){
+            layoutManager=  GridLayoutManager(this,2)
```
setting up of layout manager must be replaced into layout. Besides, in this case you will avoid any if:)

somenath1435 replied:
Actually, I'm setting number of rows for recyclerview=2 for landscape(and 1 column for potrait mode), so the function is necessary

InsanusMokrassar commented on app/src/main/java/org/xwiki/android/sync/utils/SystemTools.kt:
```diff
@@ -42,6 +44,16 @@ fun getAppVersionName(context: Context): String {
     }
 }
 
+fun getScreenOrientation(context: Context): String {
```
I don't think, that we need this function

InsanusMokrassar commented on app/src/main/res/drawable/ic_save_black_24dp.xml:
```diff
@@ -4,6 +4,6 @@
         android:viewportHeight="24.0">
     <path
-        android:fillColor="#FFFFFFFF"
+        android:fillColor="#FF000000"
```
It must be white

InsanusMokrassar replied:
Besides, I think, we must modify `accent` color inside `colors.xml`, because I think that it is better to make this button blue or, maybe, a little bit different with `primary` and `primary_dark`

InsanusMokrassar commented:
Currently it is crashing when you start set up sync settings after signing in with portrait orientation and change it in landscape (rotate device). Please, fix it, @somenath1435

somenath1435 committed:
changed fab color

somenath1435 committed:
fixed crash issue while rotating device

somenath1435 commented:
@InsanusMokrassar I've fixed crash issue and changed UI

![Screenshot_20200317-023202 1](https://user-images.githubusercontent.com/43093724/76802270-5c3b5300-67fd-11ea-9c3e-8588c6575472.png)

InsanusMokrassar committed:
Merge remote-tracking branch 'upstream/master' into syncSettingsActivityLandscape

InsanusMokrassar committed:
fixes

InsanusMokrassar committed:
remove redundant getOrientation fun

InsanusMokrassar committed:
Merge remote-tracking branch 'upstream/master' into syncSettingsActivityLandscape

InsanusMokrassar committed:
Merge remote-tracking branch 'upstream/master' into syncSettingsActivityLandscape

somenath1435 commented:
@InsanusMokrassar what about this PR?

InsanusMokrassar committed:
Merge remote-tracking branch 'upstream/master' into syncSettingsActivityLandscape

InsanusMokrassar committed:
final fixes
